wordpress-mobile/WordPress-Android
# Update more in-app Stats texts to always use western arabic numerals

Fixes `20.8` beta-testing issues discovered on #17160 (which fixes partially #17002) [internal ref: p5T066-3Ak-p2#comment-13583]

Included fixes (see previews):

1. "Total"-type cards: (I.e. Total Likes, Total Comments): Stats → Insights → Total Likes
2. Dates in Line Chart: Stats → Insights → Views & Visitors
3. "Value change" text above bar chart: Stats → Days / Weeks / Months ...

To test:

##### 1️⃣ Test Case 1 : Jetpack / Wordpress App Stats

1. Open the JP/WP App, Go to `Me` → `App Settings` → `Interface Language` and select `Arabic`.
2. Go back to `My Site` → tap `Stats` & explore the first tab (`Insights`)
   - **Expect** all updated texts to use western arabic numbers (see preview 1 & 2).
4. Go to any other `Stats` tab (Days/Weeks/...)
   - **Expect** The "value change" text to use western arabic numbers (see preview 3).

## Previews

##### 1. Text in "Total"-type cards

| Before | After |
| - | - |
| <img src="https://user-images.githubusercontent.com/4588074/192547510-7a98898b-fb46-4335-9a74-7e97c367bf5e.png" width="314" /> | <img src="https://user-images.githubusercontent.com/4588074/192547521-5ad8b835-f4a7-4dd1-94ad-ea082793cb11.png" width="314" /> |

##### 2. Views & Visitors chart dates

| Before | After |
| - | - |
| <img src="https://user-images.githubusercontent.com/4588074/192548196-5adc1521-9aa7-420a-a468-c4bf7b13a5ea.png" width="314" /> | <img src="https://user-images.githubusercontent.com/4588074/192548197-832eb00d-952c-4fb8-8783-31288d6ba013.png" width="314" /> |

##### 3. Value "change" text

| Before | After |
| - | - |
| <img src="https://user-images.githubusercontent.com/4588074/192548464-aead30bd-cca3-4275-8761-0af3e0ff1b70.png" width="314" /> | <img src="https://user-images.githubusercontent.com/4588074/192548468-f9a64e4f-9054-4692-9372-a515b8f0f922.png" width="314" /> |

## Regression Notes
1. Potential unintended areas of impact
   Texts using text view component that forces western arabic numerals.
   Updated texts for non-arabic languages.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing — the changes are rather trivial, just exposing a former private method public and reusing it.
   In one instance (for the "value change" text) we're forcing LTR text direction — this text is expected to include only numbers and symbols.

3. What automated tests I added (or what prevented me from doing so)
   N/a — PR consists of small UI changes.

PR submission checklist:
